apis/externalsecrets/v1beta1/secretstore_vault_types.go

```diff
@@ -237,3 +273,22 @@ type VaultCertAuth struct {
 	// authenticate with Vault using the Cert authentication method
 	SecretRef esmeta.SecretKeySelector `json:"secretRef,omitempty"`
 }
+
+// VaultIamAuth authenticates with Vault using the Vault's AWS IAM authentication method. Refer: https://developer.hashicorp.com/vault/docs/auth/aws
+type VaultIamAuth struct {
+
+	// Path where the AWS auth method is enabled in Vault, e.g: "aws"
+	Path string `json:"path,omitempty"`
+	// AWS region
+	Region string `json:"region,omitempty"`
+	// Vault Role. In vault, a role describes an identity with a set of permissions, groups, or policies you want to attach a user of the secrets engine
+	Role string `json:"vaultRole"`
```

Comment:
besides vaultRole, having the ability to also perform sts:AssumeRole before calling vault would prove useful. This way static creds could work without depending on a sessionToken. right now, the need for a sessionToken is kinda mandatory, otherwise vault would need to be configured directly attached to the user.

This would also allow for better support in cross-account login.